tests/test_session.py

```diff
@@ -500,6 +500,10 @@ def test_put(docker_env):
                                                             port=remotehost_port,
                                                             username='user1',
                                                             password='password1')
+    remotehost_session2 = gateway_session.get_remote_session(host=tests_util.get_host_ip(),
```

Comment:
note that doing that, `remotehost_session2` is exactly the same instance than `remotehost_session1` as gateway_session is keeping trace of all existing remote sessions and just return an existing one when matching the input criteria (user, host, port)

Reply:
to validate your fix, you can simply close the session before calling `put`

Reply:
Thanks - I just updated the test.